File: src_6_9/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

//colcon build --packages-select my_cpp_pkg
class MyNode : public rclcpp::Node{

public:
    MyNode():Node("my_first_cpp_node"){
        RCLCPP_INFO(this->get_logger(),"Hello");
        timer_ = this->create_wall_timer(
            std::chrono::seconds(1),
            std::bind(&MyNode::timer_callback, this)
        );
    }
private:
    void timer_callback(){
        RCLCPP_INFO(this->get_logger(),"Hello timer %d", counter_);
        counter_++;
    }
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_=0;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    // auto node = rclcpp::Node::make_shared("my_first_cpp_node");
    auto node = std::make_shared<MyNode>();
    //RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Hello world");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src_6_9/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3


#ros2 pkg create --build-type ament_python my_py_pkg --dependencies rclpy //src/
#colcon build --packages-select my_py_pkg
import rclpy
from rclpy.node import Node

class myNode(Node):
    def __init__(self):
        super().__init__("py_test")
        self.counter=0
        self.get_logger().info("Hello ROS2 Python")
        self.timer=self.create_timer(1.0,self.timer_callback)
    def timer_callback(self):
        self.get_logger().info("hello"+str(self.counter))
        self.counter+=1
def main(args=None):
    rclpy.init(args=args)
    # node=Node("py_test")
    # node.get_logger().info(node.get_name())

    node=myNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()